src/lamplib/src/genny/tasks/auto_tasks_all.py

```diff
@@ -0,0 +1,60 @@
+from typing import List
+import os
+import yaml
+import structlog
+from shrub.config import Configuration
+
+from genny.tasks.auto_tasks import (
+    CurrentBuildInfo,
+    WorkloadLister,
+    Repo,
+    ConfigWriter,
+    YamlReader,
+)
+
+SLOG = structlog.get_logger(__name__)
+
+def get_all_builds(global_expansions: dict[str, str], project_file_path: str) -> List[CurrentBuildInfo]:
+    """
+    Attempts to compute the expansions 
+    """
+    with open(project_file_path) as project_file:
+        project = yaml.safe_load(project_file)
+    variants = project["buildvariants"]
+    all_builds = []
+    for variant in variants:
+        if "expansions" in variant and "mongodb_setup" in variant["expansions"]:
```

Comment:
Based on https://github.com/mongodb/genny/blob/593b1f4eac70c901dabbd45bd207b846a76b7203/docs/using.md#configuring-autorun I think you need to check more than just mongodb_setup. I think you have to check all possible settings. 

Oh, maybe this is sufficient -- you are just getting the list of build variants here, and all the relevant build variants will have a mongodb_setup. NVM. Maybe worth a comment in the code. 

Also, please update that documentation when putting this in production.

Reply:
I wasn't sure if it would be a problem if someone made a generated task without a "When" block. In practice I think every task has a "when" on mongodb setup so maybe it wouldn't come up but this seemed moderately safer?